src/twim_config.rs

```diff
@@ -0,0 +1,72 @@
+use std::clone;
+use convert_case::{Case, Casing};
+use serde::{Deserialize, Serialize};
+
+use crate::{bot::Bot, bridge::Bridge};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Hash, Default)]
+pub struct Project {
+    pub emoji: String,
+    pub name: String,
+    pub title: String,
+    pub description: String,
+    pub website: String,
+    pub default_section: String,
+    pub usual_reporters: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Hash, Default)]
+pub struct Section {
+    pub emoji: String,
+    pub name: String,
+    pub title: String,
+    pub order: u32,
+    pub usual_reporters: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Config {
+    pub bot_user_id: String,
+    pub reporting_room_id: String,
+    pub admin_room_id: String,
+    pub notice_emoji: String,
+    pub image_markdown: String,
+    pub video_markdown: String,
+    pub verbs: Vec<String>,
+    pub update_config_command: String,
+    pub editors: Vec<String>,
+    pub sections: Vec<Section>,
+    pub projects: Vec<Project>,
+}
+
+impl From<&Bot> for Project {
+    fn from(bot: &Bot) -> Self {
+        Project {
+            emoji: format!("{}?", bot.title.to_case(Case::Kebab)),
+            name: bot.title.to_case(Case::Kebab),
+            title: bot.title.clone(), 
+            description: bot.description.clone(), 
+            website: bot.home.clone().unwrap_or("".to_string()), // home if not null, or repo, or empty
```

Comment:
I'd like `website` to be `home` if it's not a None, or `repository` if it's not a None, or to be an empty string if both are None.

Does Rust have some magic trickery to make a readable one-liner for this?

Reply:
`rustfmt` will likely spread this over multiple lines, but

```suggestion
            website: bot.home.clone().or_else(|| bot.repository.clone()).unwrap_or_default(),
```